runtime/queries/cpp/highlights.scm

```diff
@@ -12,56 +10,85 @@
 (template_method
   name: (field_identifier) @function)
 
-(template_function
-  name: (identifier) @function)
-
 (function_declarator
   declarator: (qualified_identifier
     name: (identifier) @function))
 
 (function_declarator
   declarator: (qualified_identifier
-    name: (identifier) @function))
+    name: (qualified_identifier
+      name: (identifier) @function)))
 
 (function_declarator
   declarator: (field_identifier) @function)
 
 ; Types
 
-((namespace_identifier) @type
- (#match? @type "^[A-Z]"))
-
+(namespace_identifier) @namespace
```

Comment:
We can separate out namespaces from types and add some stanzas for some uses:

```scm
; Namespaces

(namespace_definition name: (identifier) @namespace)
(using_declaration (identifier) @namespace)
(namespace_identifier) @namespace

; Types

(auto) @type
```

I think we can also add

```scm
(qualified_identifier name: (identifier) @type)
```

to the types section for namespaced enum variants

Reply:
These seem good, though I'd also suggest what I mentioned below about the key being @type.enum.variant instead of @type for the qualified_identifier stanza

Reply:
`(using_declaration (identifier) @namespace)` is not correct, it will match `using bar` where bar is either a typename or a value.

The following works for me:
```
(using_declaration ("using" "namespace" (identifier) @namespace))
(using_declaration ("using namespace" (qualified_identifier name: (identifier) @namespace)))
```